cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -1,19 +1,80 @@
 package kubecontroller
 
 import (
+	"net"
+	"strconv"
+	"strings"
+
 	"github.com/Azure/azure-container-networking/cns"
 	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
 )
 
-// CRDStatusToNCRequest translates a crd status to network container request
+// CRDStatusToNCRequest translates a crd status to createnetworkcontainer request
 func CRDStatusToNCRequest(crdStatus *nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
-	//TODO: Translate CRD status into network container request
-	//Mat will pick up from here
-	return nil, nil
+	var (
+		createNCRequest   cns.CreateNetworkContainerRequest
+		nc                nnc.NetworkContainer
+		secondaryIPConfig cns.SecondaryIPConfig
+		ipSubnet          cns.IPSubnet
+		ipAssignment      nnc.IPAssignment
+		prefixLength      int
+	)
+
+	createNCRequest.SecondaryIPConfigs = make(map[string]cns.SecondaryIPConfig)
+
+	for _, nc = range crdStatus.NetworkContainers {
+		createNCRequest.NetworkContainerid = nc.ID
+		createNCRequest.PrimaryInterfaceIdentifier = nc.PrimaryIP
+		prefixLength = maskStringToPrefixLength(nc.Netmask)
+
+		for _, ipAssignment = range nc.IPAssignments {
+			ipSubnet.IPAddress = ipAssignment.IP
+			ipSubnet.PrefixLength = uint8(prefixLength)
+			secondaryIPConfig = cns.SecondaryIPConfig{
+				IPConfig: ipSubnet,
+			}
+			createNCRequest.SecondaryIPConfigs[ipAssignment.Name] = secondaryIPConfig
+		}
+	}
+
+	//Only returning the first network container for now, later we will return a list
+	return &createNCRequest, nil
```

Comment:
Can this be addressed in this PR? Or is there something else that blocks this returning a list?

Reply:
As Neha already implemented her CreateOrUpdateNCRequest it will be a bigger change so I think we should tackle it in a separate pr